wire-library/wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt

```diff
@@ -768,15 +790,29 @@ class SwiftGenerator private constructor(
                     )
                   }
 
-                  if (field.typeName.optional) {
+                  val defaultValue = field.codableDefaultValue
+
+                  if (field.isOptional) {
+                    // A proto3 field that is defined with the optional keyword supports field presence.
+                    // Fields that have a value set and that support field presence always include the field value
+                    // in the JSON-encoded output, even if it is the default value.
                     beginControlFlow("if", "includeDefaults || self.%N != nil", field.name)
```

Comment:
We should verify the `protoc` behavior for this. The comment above from the docs doesn't actually clarify what the behavior is for *unset* optional fields. Do we encode `null`? Do we omit it? Do we do the former if `includeDefaults == true` and the latter if not (which is what's currently happening)?

The last option (the current option) seems reasonable for now, but wanted to call it out.

Reply:
Yup, I'll take a look.
JSON definitely supports an explicit `null` for any value so it's definitely _valid_.